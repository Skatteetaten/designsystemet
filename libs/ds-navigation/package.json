{
  "name": "@skatteetaten/ds-navigation",
  "version": "1.3.0",
  "groupId": "no.skatteetaten.aurora",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Skatteetaten/designsystemet.git"
  },
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "react": "^18",
    "react-i18next": "14.1.2",
    "i18next": "^23",
    "@skatteetaten/ds-buttons": "1.3.0",
    "@skatteetaten/ds-core-utils": "1.3.0",
    "@skatteetaten/ds-icons": "1.3.0",
    "@skatteetaten/ds-progress": "1.3.0"
  },
  "scripts": {},
  "dependencies": {},
  "devDependencies": {}
}
