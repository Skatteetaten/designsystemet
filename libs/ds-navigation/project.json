{
  "name": "ds-navigation",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/ds-navigation/src",
  "projectType": "library",
  "tags": [],
  "implicitDependencies": ["!ds-dev-config"],
  "targets": {
    "lint": {
      "executor": "@nx/eslint:lint",
      "outputs": ["{options.outputFile}"]
    },
    "build": {
      "executor": "@nx/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/ds-navigation",
        "tsConfig": "libs/ds-navigation/tsconfig.lib.json",
        "project": "libs/ds-navigation/package.json",
        "entryFile": "libs/ds-navigation/src/index.ts",
        "external": ["react", "react-dom", "react/jsx-runtime"],
        "rollupConfig": ["libs/ds-navigation/rollup.config.js"],
        "compiler": "swc",
        "assets": [
          {
            "glob": "libs/ds-navigation/README.md",
            "input": ".",
            "output": "."
          },
          {
            "glob": "LICENSE",
            "input": ".",
            "output": "."
          }
        ]
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/ds-navigation/jest.config.ts"
      }
    },
    "stylelint": {
      "executor": "nx-stylelint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-navigation/**/*.css",
          "libs/ds-navigation/**/*.scss"
        ]
      }
    },
    "tsc": {
      "executor": "@skatteetaten/ds-dev-config:tsCheck",
      "options": {
        "tsConfig": ["tsconfig.lib.json", "tsconfig.spec.json"]
      },
      "dependsOn": [
        {
          "projects": "ds-core-designtokens",
          "target": "build"
        }
      ]
    },
    "set-version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm version {args.packageVersion} --no-git-tag",
        "cwd": "libs/ds-navigation"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm publish dist/libs/ds-navigation {args.switches}"
      }
    }
  }
}
