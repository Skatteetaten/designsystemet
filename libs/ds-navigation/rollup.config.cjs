const path = require('path');

const createRollupConfig = require('../ds-dev-config/build/rollup-config-factory.cjs');

const srcDir = path.resolve(__dirname, `src`);
const pathToTsConfig = 'libs/ds-navigation/tsconfig.lib.json';
const pathToCSS = path.resolve(__dirname, `**/*.{css,scss}`);
const outputDir = path.resolve('./', `dist/libs/ds-navigation/`);

module.exports = (config) =>
  createRollupConfig(config, srcDir, pathToTsConfig, pathToCSS, outputDir);
