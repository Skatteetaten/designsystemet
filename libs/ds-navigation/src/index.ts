export * from './Breadcrumbs/Breadcrumbs';
export * from './Breadcrumbs/Breadcrumbs.types';
export * from './BreadcrumbItem/BreadcrumbItem';
export * from './BreadcrumbItem/BreadcrumbItem.types';
export * from './BreadcrumbLink/BreadcrumbLink';
export * from './BreadcrumbLink/BreadcrumbLink.types';
export * from './NavigationTile/NavigationTile';
export * from './NavigationTile/NavigationTile.types';
export * from './Pagination/Pagination';
export * from './Pagination/Pagination.types';
