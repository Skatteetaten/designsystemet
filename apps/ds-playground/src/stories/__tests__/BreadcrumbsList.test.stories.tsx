import { Breadcrumbs } from '@skatteetaten/ds-navigation';
import { Meta, StoryObj } from '@storybook/react';
import { expect, within } from '@storybook/test';

const elementId = 'htmlId';

const verifyAttribute =
  (attribute: string, expectedValue: string) =>
  async ({ canvasElement }: { canvasElement: HTMLElement }): Promise<void> => {
    const canvas = within(canvasElement);
    await expect(canvas.getByRole('list')).toHaveAttribute(
      attribute,
      expectedValue
    );
  };

const meta = {
  component: Breadcrumbs.List,
  title: 'Tester/Breadcrumbs/List',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    children: { table: { disable: true } },
    shouldCollapse: { table: { disable: true } },
    showLastItemAsCurrentPage: { table: { disable: true } },
  },
  args: {
    children: [
      <Breadcrumbs.Item key={'i1'}>
        <Breadcrumbs.Link href={'#'}>
          {'Bedrift og organisasjon'}
        </Breadcrumbs.Link>
      </Breadcrumbs.Item>,
      <Breadcrumbs.Item key={'i2'}>
        <Breadcrumbs.Link href={'#'}>{'Avgifter'}</Breadcrumbs.Link>
      </Breadcrumbs.Item>,
      <Breadcrumbs.Item key={'i3'}>
        <Breadcrumbs.Link href={'#'}>
          {'Avgifter på bil og kjøretøy'}
        </Breadcrumbs.Link>
      </Breadcrumbs.Item>,
      <Breadcrumbs.Item key={'i4'}>
        <Breadcrumbs.Link href={'#'}>{'Vektårsavgift'}</Breadcrumbs.Link>
      </Breadcrumbs.Item>,
    ],
  },
} satisfies Meta<typeof Breadcrumbs.List>;
export default meta;
type Story = StoryObj<typeof meta>;

export const WithRef = {
  name: 'With Ref (FA1)',
  args: {
    ref: (instance: HTMLElement | null): void => {
      if (instance) {
        instance.id = 'dummyIdForwardedFromRef';
      }
    },
  },
  argTypes: {
    ref: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: verifyAttribute('id', 'dummyIdForwardedFromRef'),
} satisfies Story;

export const WithAttributes = {
  name: 'With Attributes(FA2-5)',
  args: {
    id: elementId,
    className: 'dummyClassname',
    lang: 'nb',
    'data-testid': '123ID',
  },
  argTypes: {
    id: { table: { disable: false } },
    className: { table: { disable: false } },
    lang: { table: { disable: false } },
    'data-testid': { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const container = canvas.getByRole('list');
    await expect(container).toHaveClass('dummyClassname');
    await expect(container).toHaveAttribute('id', elementId);
    await expect(container).toHaveAttribute('lang', 'nb');
    await expect(container).toHaveAttribute('data-testid', '123ID');
  },
} satisfies Story;

export const WithoutIsShowLastItemAsCurrentPage = {
  name: 'Without Show Last Item As Current Page (B3)',
  args: {
    showLastItemAsCurrentPage: false,
  },
  argTypes: { showLastItemAsCurrentPage: { table: { disable: false } } },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const listItems = canvas.getAllByRole('listitem');
    const lastItem = listItems[listItems.length - 1];
    await expect(lastItem).not.toHaveAttribute('aria-current', 'page');
  },
} satisfies Story;
