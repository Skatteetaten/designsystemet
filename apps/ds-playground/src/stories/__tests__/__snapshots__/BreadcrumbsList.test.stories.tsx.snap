// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/Breadcrumbs/List WithAttributes play-test 1`] = `
<div data-test-block="true">
  <ol id="htmlId"
      lang="nb"
      data-testid="123ID"
      class="_breadcrumbsList_p1lxe_1   dummyClassname"
  >
    <li class="_breadcrumbsItem_1jlx4_1">
      <a class="_link_1y34f_1  "
         href="#"
      >
        Bedrift og organisasjon
      </a>
      <span class="_breadcrumbsItemSeparator_1jlx4_10"
            aria-hidden="true"
      >
        /
      </span>
    </li>
    <li class="_breadcrumbsItem_1jlx4_1">
      <a class="_link_1y34f_1  "
         href="#"
      >
        Avgifter
      </a>
      <span class="_breadcrumbsItemSeparator_1jlx4_10"
            aria-hidden="true"
      >
        /
      </span>
    </li>
    <li class="_breadcrumbsItem_1jlx4_1">
      <a class="_link_1y34f_1  "
         href="#"
      >
        Avgifter på bil og kjøretøy
      </a>
      <span class="_breadcrumbsItemSeparator_1jlx4_10"
            aria-hidden="true"
      >
        /
      </span>
    </li>
    <li class="_breadcrumbsItem_1jlx4_1">
      <a class="_link_1y34f_1  "
         href="#"
      >
        Vektårsavgift
      </a>
      <span class="_breadcrumbsItemSeparator_1jlx4_10"
            aria-hidden="true"
      >
        /
      </span>
    </li>
  </ol>
</div>
`;

exports[`Tester/Breadcrumbs/List WithRef play-test 1`] = `
<div data-test-block="true">
  <ol class="_breadcrumbsList_p1lxe_1"
      id="dummyIdForwardedFromRef"
  >
    <li class="_breadcrumbsItem_1jlx4_1">
      <a class="_link_1y34f_1  "
         href="#"
      >
        Bedrift og organisasjon
      </a>
      <span class="_breadcrumbsItemSeparator_1jlx4_10"
            aria-hidden="true"
      >
        /
      </span>
    </li>
    <li class="_breadcrumbsItem_1jlx4_1">
      <a class="_link_1y34f_1  "
         href="#"
      >
        Avgifter
      </a>
      <span class="_breadcrumbsItemSeparator_1jlx4_10"
            aria-hidden="true"
      >
        /
      </span>
    </li>
    <li class="_breadcrumbsItem_1jlx4_1">
      <a class="_link_1y34f_1  "
         href="#"
      >
        Avgifter på bil og kjøretøy
      </a>
      <span class="_breadcrumbsItemSeparator_1jlx4_10"
            aria-hidden="true"
      >
        /
      </span>
    </li>
    <li class="_breadcrumbsItem_1jlx4_1">
      <a class="_link_1y34f_1  "
         href="#"
      >
        Vektårsavgift
      </a>
      <span class="_breadcrumbsItemSeparator_1jlx4_10"
            aria-hidden="true"
      >
        /
      </span>
    </li>
  </ol>
</div>
`;
